p2p/src/peer.rs

```diff
@@ -53,17 +53,8 @@ pub struct Peer {
 	pub info: PeerInfo,
 	state: Arc<RwLock<State>>,
 	// set of all hashes known to this peer (so no need to send)
-	tracking_adapter: TrackingAdapter,
-	connection: Option<Mutex<conn::Tracker>>,
-}
-
-macro_rules! connection {
-	($holder:expr) => {
-		match $holder.connection.as_ref() {
-			Some(conn) => conn.lock(),
-			None => return Err(Error::ConnectionClose),
-			}
-	};
+	tracking_adapter: Arc<TrackingAdapter>,
```

Comment:
Do we really need `Arc` here and below? Would be nice to get rid of this indirection (we didn't have it before)

Reply:
I added this for consistency with peer state above but I agree its not necessary.
Let me take another pass over this.